custom_components/tuya_local/devices/madimack_elite_v3_heatpump_updated.yaml

```diff
@@ -0,0 +1,200 @@
+name: Madimack Elite V3 pool heatpump
+primary_entity:
+  entity: climate
+  dps:
+    - id: 101
+      name: hvac_mode
+      type: boolean
+      mapping:
+        - dps_val: false
+          value: "off"
+          icon: "mdi:hvac-off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: auto
+              icon: "mdi:refresh-auto"
+              value: auto
+            - dps_val: cold
+              icon: "mdi:snowflake"
+              value: cool
+            - dps_val: heating
+              icon: "mdi:hot-tub"
+              value: heat
+    - id: 102
+      name: mode
+      type: string
+      hidden: true
+    - id: 104
+      name: temperature
+      type: integer
+      mapping:
+        - constraint: temperature_unit
+          conditions:
+            - dps_val: f
+              range:
+                min: 60
+                max: 104
+      range:
+        min: 18
+        max: 40
+    # ^ in reality, the range is different for cool and auto
+    - id: 105
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: silence
+          icon: "mdi:weather-night"
+          value: sleep
+        - dps_val: perfect
+          icon: "mdi:weather-sunny"
+          value: comfort
+        - dps_val: power
+          icon: "mdi:sun-thermometer"
+          value: boost
+    - id: 106
+      name: temperature_unit

```

Comment:
Is this also settable?  There should be a select entity for setting it, if so.

Reply:
This is how I found it in the original version. The dp is writeable - I'll add a select primary entity for this

Reply:
I have added (as a secondary entity) a select switch for the units. It does successfully change the units from celsius to fahrenheit in the backend of tuya, changing the set temps and temperature limits accordingly.

This is difficult to test in home assistant though - the resulting climate card looks like it is being translated by HA from F to C, as the temperatures listed don't change. I suspect the system settings over-ride the units being used.

It may be best if I left this select switch undefined - all the remaining temperature sensors pull this DP to ensure the correct units, and HA converts to the system settings? What would you prefer?

Reply:
You are correct. Home Assistant translates the temperatures in the frontend to the user's preferred units.  So if the backend is working correctly, the displayed value should not change.  If the backend is not doing things correctly, then you would see the values change but not match the units that are displayed in the frontend.